test/test_prototype_transforms.py

```diff
@@ -2355,3 +2356,124 @@ def test_detection_preset(image_type, label_type, data_augmentation, to_tensor):
 
     out["label"] = torch.tensor(out["label"])
     assert out["boxes"].shape[0] == out["masks"].shape[0] == out["label"].shape[0] == num_boxes
+
+
+@pytest.mark.parametrize("min_size", (1, 10))
+@pytest.mark.parametrize(
+    "labels_getter", ("default", "labels", lambda inputs: inputs["labels"], None, lambda inputs: None)
+)
+@pytest.mark.parametrize("batch", (True, False))
+def test_sanitize_bounding_boxes(min_size, labels_getter, batch):
+    H, W = 256, 128
+
+    boxes_and_validity = [
+        ([0, 1, 10, 1], False),  # Y1 == Y2
+        ([0, 1, 0, 20], False),  # X1 == X2
+        ([0, 0, min_size - 1, 10], False),  # H < min_size
+        ([0, 0, 10, min_size - 1], False),  # W < min_size
+        ([0, 0, 10, H + 1], False),  # Y2 > H
+        ([0, 0, W + 1, 10], False),  # X2 > W
+        ([-1, 1, 10, 20], False),  # any < 0
+        ([0, 0, -1, 20], False),  # any < 0
+        ([0, 0, -10, -1], False),  # any < 0
+        ([0, 0, min_size, 10], True),  # H < min_size
+        ([0, 0, 10, min_size], True),  # W < min_size
+        ([0, 0, W, H], True),  # TODO: Is that actually OK?? Should it be -1?
+        ([1, 1, 30, 20], True),
+        ([0, 0, 10, 10], True),
+        ([1, 1, 30, 20], True),
+    ]
+
+    random.shuffle(boxes_and_validity)  # For test robustness: mix order of wrong and correct cases
+    boxes, is_valid_mask = zip(*boxes_and_validity)
+    valid_indices = [i for (i, is_valid) in enumerate(is_valid_mask) if is_valid]
+
+    boxes = torch.tensor(boxes)
+    labels = torch.arange(boxes.shape[-2])
+
+    if batch:
+        boxes = boxes[None]
+        labels = labels[None]
+        valid_indices = [valid_indices]
+
+    boxes = datapoints.BoundingBox(
+        boxes,
+        format=datapoints.BoundingBoxFormat.XYXY,
+        spatial_size=(H, W),
+    )
+
+    sample = {
+        "image": torch.randint(0, 256, size=(1, 3, H, W), dtype=torch.uint8),
+        "labels": labels,
+        "boxes": boxes,
+        "whatever": torch.rand(10),
+        "None": None,
+    }
+
+    out = transforms.SanitizeBoundingBoxes(min_size=min_size, labels_getter=labels_getter)(sample)
+
+    assert out["image"] is sample["image"]
+    assert out["whatever"] is sample["whatever"]
+
+    if labels_getter is None or (callable(labels_getter) and labels_getter({"labels": "blah"}) is None):
```

Comment:
I see that returning `None` from callable labels getter is intentional. Why though? If someone explicitly sets this, why would they want to (silently) not transform labels? Imagine, someone just forgets a `return` statement (which happens regularly but not often to me).

Reply:
what if the label is from the image and not the boxes?

Reply:
They should pass `labels=None`? Unless I miss something, this only applies to user defined callables, since the default heuristic errors in case it doesn't find anything. So why would a user use `labels=lambda: None` if that has the same behavior as `labels=None`? The only reason I see is if the callable can conditionally return `None` or the labels. But I fail to see a use case for that.

Reply:
what if they have some samples with labels and some without?

Reply:
imagine the samples are

```
labels_getter inputs = inputs[2]
# train
(img, bboxes, labels)
# test? Or some other weird stuff??
(img, bboxes, None)
```
?

Reply:
WE discussed offline and I just understood the mechanism wrong. Everything good as is.